core/src/kvs/surrealcs/mod.rs

```diff
@@ -0,0 +1,426 @@
+#![cfg(feature = "kv-surrealcs")]
+
+use crate::err::Error;
+use crate::key::debug::Sprintable;
+use crate::kvs::Check;
+use crate::kvs::Key;
+use crate::kvs::Val;
+use futures::lock::Mutex;
+use std::fmt::Debug;
+use std::ops::Range;
+use std::sync::Arc;
+use surrealcs::kernel::messages::server::interface::ServerTransactionMessage;
+use surrealcs::kernel::messages::server::kv_operations::*;
+use surrealcs::router::create_connection_pool;
+use surrealcs::transactions::interface::bridge::BridgeHandle;
+use surrealcs::transactions::interface::interface::{
+	Any as AnyState, Transaction as SurrealCSTransaction,
+};
+
+/// The main struct that is used to interact with the database.
+#[derive(Clone)]
+#[non_exhaustive]
+pub struct Datastore {}
+
+#[non_exhaustive]
+pub struct Transaction {
+	/// Is the transaction complete?
+	done: bool,
+	/// Is the transaction writeable?
+	write: bool,
+	/// Should we check unhandled transactions?
+	check: Check,
+	/// Has the transaction been started?
+	started: bool,
+	/// The underlying datastore transaction
+	inner: Arc<Mutex<SurrealCSTransaction<AnyState>>>,
+}
+
+impl Drop for Transaction {
+	fn drop(&mut self) {
+		if !self.done && self.write {
+			// Check if already panicking
+			if std::thread::panicking() {
+				return;
+			}
+			// Handle the behaviour
+			match self.check {
+				Check::None => {
+					trace!("A transaction was dropped without being committed or cancelled");
+				}
+				Check::Warn => {
+					warn!("A transaction was dropped without being committed or cancelled");
+				}
+				Check::Panic => {
+					#[cfg(debug_assertions)]
+					{
+						let backtrace = std::backtrace::Backtrace::force_capture();
+						if let std::backtrace::BacktraceStatus::Captured = backtrace.status() {
+							println!("{}", backtrace);
+						}
+					}
+					panic!("A transaction was dropped without being committed or cancelled");
+				}
+			}
+		}
+	}
+}
+
+impl Datastore {
+	/// Open a new database
+	pub(crate) async fn new(path: &str) -> Result<Datastore, Error> {
+		create_connection_pool(path, None).await.unwrap();
```

Comment:
this looks like global state, what happens if you try to create multiple Datastores?